api/scraper.py

```diff
@@ -0,0 +1,351 @@
+    deal.url = f"https://mediamarkt.ch{api_response['1']['url']}"
+    deal.color = "#E20000"
+    deal.availability = "Nur solange Vorrat"
+    deal.old_price = api_response["1"]["oldPrice"]
+    deal.new_price = api_response["1"]["price"]["price"]
+
+    return deal
+
+
+def get_any_deal(deal):
+    """
+    Return deal by string
+    """
+    logging.info("Getting deal for %s", deal)
+    try:
+        if deal == "digitec":
+            return get_digitec_deal("https://www.digitec.ch/de/daily-deal")
+        if deal == "galaxus":
+            return get_digitec_deal("https://www.galaxus.ch/de/daily-deal", "#222")
+        if deal == "daydeal_tagesdeal":
+            return get_daydealbeta("tagesdeal")
+        if deal == "daydeal_it_multimedia":
+            return get_daydealbeta("it-multimedia")
+        if deal == "daydeal_haushalt_wohnen":
+            return get_daydealbeta("haushalt-wohnen")
+        if deal == "daydeal_supermarkt_drogerie":
+            return get_daydealbeta("supermarkt-drogerie")
+        if deal == "daydeal_baumarkt_hobby":
+            return get_daydealbeta("baumarkt-hobby")
+        if deal == "daydeal_familie_baby":
+            return get_daydealbeta("familie-baby")
+        if deal == "daydeal_sport_freizeit":
+            return get_daydealbeta("sport-freizeit")
+        if deal == "daydeal_daily":
+            return get_deal("https://www.daydeal.ch/")
+        if deal == "daydeal_weekly":
+            return get_deal("https://www.daydeal.ch/deal-of-the-week/")
+        if deal == "mediamarkt":
+            return get_mediamarkt_deal("https://www.mediamarkt.ch")
+        if deal == "zmin":
+            return get_zmin_deal("angebot-des-tages")
+        if deal == "zmin_weekly":
+            return get_zmin_deal("wochenangebot")
+        if deal == "blick":
+            return get_blick_deal("https://box.blick.ch/deal-des-tages")
+        if deal == "blick_weekly":
+            return get_blick_deal("https://box.blick.ch/deal-der-woche")
+        return {}
+
+    except Exception as ex:
+        logging.error("Failed to parse %s", deal)
+        logging.error("The following exception was thrown:\n%s", ex)
+        return []
+
+
+def scrape():
+    logging.basicConfig(
+        format="%(levelname)s %(asctime)s - %(message)s", level=logging.INFO
+    )
+
+    deals_list = [
+        "digitec",
+        "galaxus",
+        "daydeal_tagesdeal",
+        "daydeal_it_multimedia",
+        "daydeal_haushalt_wohnen",
+        "daydeal_supermarkt_drogerie",
+        "daydeal_familie_baby",
+        "daydeal_baumarkt_hobby",
+        "daydeal_sport_freizeit",
+        "blick",
+        "blick_weekly",
+        "mediamarkt",
+        "zmin",
+        "zmin_weekly",
+    ]
+
+
+    with Pool(5) as p:
+        deals = p.map(get_any_deal, deals_list)
+
+
+    output = {}
+    for i, _ in enumerate(deals_list):
+        if not deals[i] == []:
+            output[deals_list[i]] = json.loads(str(deals[i]))
```

Comment:
this can be refactored:
 
```py
output = {}
for deal_name, deal in zip(deals_list, deals):
   if not deal == []:
       output[deal_name] = json.loads(deal)
 ```

and then it can be further refactored by using a dict comprehension like this:
```py
output = {deal_name: json.loads(deal) for deal_name, deal in zip(deals_list, deals) if not deal == []}
```

then you should be able to remove `if not deal == []` as the get_deal functions all return a `Deal`

```py
output = {deal_name: json.loads(deal) for deal_name, deal in zip(deals_list, deals)}
```

```suggestion
output = {deal_name: json.loads(deal) for deal_name, deal in zip(deals_list, deals)}
```